### PR TITLE
closet skeleton event

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

random event where a ghost role skeleton spawns in a closed closet,

why? Why not, rather! it's fun and i love skelebones.

also did some tweaks to skeleton skin tones so they aren't black anymore.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Skeletons have been reported in the closets of various station officials. And maintenance.
- tweak: Skeletons can no longer have a pitch-black coloration.

### CODE REVIEW

**Comment by UbaserB** [comment]
will this skeleton be a ghost role?

**Comment by EmoGarbage404** [comment]
of course

**Comment by LankLTE** [comment]
Can’t wait to spawn in the Captain’s locked locker, then have to break my way out of the locker just to be stuck in his office

**Comment by Doru991** [comment]
THE SKELETON APPEARS

File: Resources/Prototypes/GameRules/events.yml [hunk 1]
```diff
@@ -55,6 +55,18 @@
...
 
+- type: entity
+  parent: BaseGameRule
+  id: ClosetSkeleton
+  noSpawn: true
+  components:
+  - type: StationEvent
+    weight: 5
+    duration: 1
+    earliestStart: 30
```
Comment (mirrorcult) [inline Resources/Prototypes/GameRules/events.yml:66]: why earliest start 30 i dont think it matters

File: Content.Server/StationEvents/Events/RandomClosetSpawnRule.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        var query = EntityQueryEnumerator<EntityStorageComponent, TransformComponent>();
+        while (query.MoveNext(out var ent, out var storage, out var xform))
+        {
+            if (StationSystem.GetOwningStation(ent, xform) != station)
+                continue;
+
+            if (!_entityStorage.CanInsert(ent, storage) || storage.Open)
+                continue;
+
+            var spawn = Spawn(comp.Prototype, xform.Coordinates);
```
Comment (deltanedas) [inline Content.Server/StationEvents/Events/RandomClosetSpawnRule.cs:29]: this isnt really random its just whatever order the entities are in?

File: Content.Server/StationEvents/Components/RandomClosetSpawnRuleComponent.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+/// </summary>
+[RegisterComponent, Access(typeof(RandomClosetSpawnRule))]
+public sealed partial class RandomClosetSpawnRuleComponent : Component
+{
+    /// <summary>
+    /// The entity to be spawned.
+    /// </summary>
+    [DataField("prototype", required: true, customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+    public string Prototype = string.Empty;
+}
```
Comment (deltanedas) [inline Content.Server/StationEvents/Components/RandomClosetSpawnRuleComponent.cs:18]: shouldnt this have a whitelist to select closets specifically (and be renamed)

File: Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl [hunk 4]
```diff
@@ -122,6 +122,9 @@ ghost-role-information-skeleton-pirate-description = Cause chaos and loot the st
 ghost-role-information-skeleton-biker-name = Skeleton Biker
 ghost-role-information-skeleton-biker-description = Ride around on your sweet ride.
 
+ghost-role-information-closet-skeleton-name = Closet Skeleton
+ghost-role-information-closet-skeleton-description = Exist as a primordial force on the station. Live happily with the crew or wage sweet skeletal war. You have no allegiance.
```
Comment (deltanedas) [inline Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl:126]: "wage sweet skeletal war" might be okaying antagonist activity so it should say you can be an antagonist if you choose

**Review by deltanedas — CHANGES_REQUESTED** [review]
accidentally left comment for first one pro

**Comment by Cvasic** [comment]
Spooky scary skeletons

**Comment by JoeHammad1844** [comment]
are these hostiles and pick up items like the spirate and salvager AI?

**Comment by potato1234x** [comment]
This should be merged for halloween as well as skeletons being enabled round start

**Comment by EmoGarbage404** [comment]
> This should be merged for halloween as well as skeletons being enabled round start

it's spooky month all year round in this codebase

File: Content.Server/StationEvents/Events/RandomEntityStorageSpawnRule.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            if (!_entityStorage.CanInsert(ent, storage) || storage.Open)
+                continue;
+
+            var spawn = Spawn(comp.Prototype, xform.Coordinates);
+            if (!_entityStorage.Insert(spawn, ent, storage))
+            {
+                Del(spawn);
+            }
+            return;
+        }
```
Comment (mirrorcult) [inline Content.Server/StationEvents/Events/RandomEntityStorageSpawnRule.cs:35]: this isnt random

File: Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl (hunk 4 above)
Comment (mirrorcult) [reply Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl:126]: yes these should be more explicit that its antag allowed imo

**Review by mirrorcult — DISMISSED** [review]

File: Content.Shared/Humanoid/SkinColor.cs [hunk 6]
```diff
@@ -131,7 +132,7 @@ public static Color TintedHues(Color color)
     public static bool VerifyTintedHues(Color color)
     {
         // tinted hues just ensures saturation is always .1, or 10% saturation at all times
-        return Color.ToHsv(color).Y != .1f;
+        return Color.ToHsl(color).Y <= .1f && Color.ToHsl(color).Y >= .85f;
```
Comment (Slava0135) [inline Content.Shared/Humanoid/SkinColor.cs:135]: can we put 0.1f and 0.85f into consts? (those numbers are used in `TintedHues` method above too)

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**